recipe/patches/0003-Remove-all-dependencies-from-setup.py.patch

```diff
@@ -77,7 +96,8 @@ index c361a6880..3dc3e5777 100644
 -        "click >= 7.0, <= 8.0.4",
 -        "dataclasses; python_version < '3.7'",
 -        "filelock",
--        "grpcio >= 1.28.1, <= 1.43.0",
+-        "grpcio >= 1.28.1, <= 1.43.0; python_version < '3.10'",
+-        "grpcio >= 1.42.0, <= 1.43.0; python_version >= '3.10'",
```

Comment:
This is a really restrictive condition. Conda-forge is at 1.46/1.47/1.48 and not rebuilding grpcio 1.43 against new packages like abseil & protobuf anymore. That means that ray becomes progressively less co-installable with other packages (being kept up-to-date) every day.

Reply:
I don't have much context about these changes, but the latest I could find was https://github.com/ray-project/ray/pull/27244, suggesting that we run into problems with newer versions

Reply:
Well, there's quite a gap between `<=1.43` & `>=1.48`. As it looks from https://github.com/ray-project/ray/issues/27299, this is a pretty gratuitous restriction that I think we should consider overruling (at least up until 1.46), for the reasons I outlined in my [comment](https://github.com/ray-project/ray/issues/27299#issuecomment-1240457098) there.

Reply:
As long as we avoid the version where Ray hangs this is fine with me!

Reply:
I [asked](https://github.com/ray-project/ray/issues/28354) what versions are affected.

Reply:
> I [asked](https://github.com/ray-project/ray/issues/28354) what versions are affected.

If you want we can merge this PR and update the build if necessary, once there is an answer.